Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -114,6 +115,22 @@ public void SerializesAndDeserializesSubtypesBasedOnTypeAnnotations()
             );
         }
 
+        [TestMethod]
+        public void SerializeMaxDepthException()
+        {
+            var serializer = new IonSerializer(new IonSerializationOptions {MaxDepth = 3});
+            Assert.ThrowsException<NotSupportedException>(() => serializer.Serialize(TestObjects.honda));
+        }
+        
+        [TestMethod]
+        public void DeserializeMaxDepthException()
+        {
+            var stream = new IonSerializer().Serialize(TestObjects.honda);
+            
+            var serializer = new IonSerializer(new IonSerializationOptions {MaxDepth = 3});
+            Assert.ThrowsException<NotSupportedException>(() => serializer.Deserialize<Car>(stream));
```

Comment:
This should not throw an exception, it should just ignore whatever is too deep.

Reply:
Now we just return null objects when MaxDepth is reached.